src/classes/OPP_OpportunityNaming_BATCH.cls

```diff
@@ -59,27 +76,45 @@ public class OPP_OpportunityNaming_BATCH implements Database.Batchable<sObject>,
     * @description Batch process execute method. Names and updates all opportunities in the current batch.
     */
     public void execute(Database.BatchableContext BC, List<sObject> scope) {
+        List<Opportunity> oppsToProcess = scope;
+        lastOppIdProcessed = oppsToProcess[oppsToProcess.size() - 1].Id;
+
         //save old opp names to see if we need an update
-        list<opportunity> oppsForUpdate = new list<Opportunity>();
-        map<id,string> oppNames = new map<id,string>();
-        for (Opportunity opp : (list<Opportunity>)scope)
-            oppNames.put(opp.id, opp.Name);
+        Map<Id, String> originalOppNamesById = new Map<Id, String>();
+        for (Opportunity opp : oppsToProcess) {
+            originalOppNamesById.put(opp.id, opp.Name);
+        }
 
         //refresh names
-        OPP_OpportunityNaming.refreshOppNames((list<Opportunity>)scope);
+        OPP_OpportunityNaming.refreshOppNames(oppsToProcess);
 
         //find which names have been updated, add to list
-        for (Opportunity opp : (list<Opportunity>)scope) {
-            if (opp.Name != oppNames.get(opp.id))
+        List<Opportunity> oppsForUpdate = new List<Opportunity>();
+        for (Opportunity opp : oppsToProcess) {
+            if (opp.Name != originalOppNamesById.get(opp.id)) {
                 oppsForUpdate.add(opp);
+            }
         }
 
         if (!oppsForUpdate.isEmpty()) {
-            database.update(oppsForUpdate, false);
+            Database.update(oppsForUpdate, false);
```

Comment:
Do you think we should use `UTIL_DMLService.updateRecords(oppsForUpdate, false)` instead?

Reply:
Yeah I think that makes sense, I'll make the change